src/playlist.js

```diff
@@ -407,7 +419,12 @@ export const getMediaInfoForTime = function({
     break;
   }
 
-  if (time < 0) {
+  // for normal playlists we can compare against zero
+  // but we have to be much more specific for LLHLS.
+  // as even half a second can cause us to guess the wrong segment.
+  const zeroOrPartTarget = playlist.partTargetDuration || 0;
```

Comment:
WIthout this fix we end up picking a part ahead of currentTime quite often, presumably because parts are so small and it's hard to pick one exactly like this.